aten/src/ATen/native/cuda/TensorFactories.cu

```diff
@@ -102,18 +102,31 @@ Tensor& randperm_out_cuda(Tensor& result, int64_t n, Generator* generator) {
   AT_DISPATCH_ALL_TYPES(
     result.scalar_type(), "randperm_out_cuda", [&] {
       auto keys = at::empty(result.sizes(), result.options()).random_(generator);
-
-      auto result_data = thrust::device_ptr<scalar_t>(result.data<scalar_t>());
       auto keys_data = thrust::device_ptr<scalar_t>(keys.data<scalar_t>());
 
+      // shuffled_data points to the underlying data of the output tensor if the tensor is contiguous; otherwise it
+      // points to a new tensor.
+      Tensor shuffled;
+      thrust::device_ptr<scalar_t> shuffled_data;
+      if (result.is_contiguous()) {
+        shuffled_data = thrust::device_ptr<scalar_t>(result.data<scalar_t>());
+      } else {
+        at::arange_out(shuffled, 0, n);
```

Comment:
why does this need to be arange_out and not empty_out?  I'm also not sure we guarantee the behavior of passing in an undefined tensor here -- shouldn't we just call empty_like of result?

Reply:
Right. Thanks for the catch -- I missed this when doing a previous update.

Reply:
It needs to be `arange` because we need to shuffle it later.